### PR TITLE
GDB-7732: Added change repository listener and dirty check functionality

### PR DESCRIPTION
**Pull request by boyan-tonchev**
What
Added functionality to prevent changing the repository or location without user confirmation.

Why
- Similarity indexes are specific to each repository, so when a similarity index is open for editing, the repository should not be changed. If this occurs, the user should be notified.
- If a similarity index is open for editing, or if the create similarity index form is open and the user makes changes in the form but tries to change the location without saving the changes, they should be notified about the unsaved changes.

How
Added a listener for changes in the repository. Added a listener for changes in the location. Implemented confirmation prompts if any of these events occur.

### CODE REVIEW

File: src/js/angular/similarity/controllers/create-index.controller.js [hunk 1]
```diff
@@ -703,11 +836,11 @@ function CreateSimilarityIdxCtrl($scope, toastr, $uibModal, $timeout, Similarity
...
             Promise.all([SimilarityRestService.getSearchQueries(), SimilarityRestService.getSamples(), $repositories.getPrefixes(activeRepo.id)])
-                .then(([searchQueries, samples, usedPrefixes]) => {
+                .then(([searchQueriesResponses, samples, usedPrefixes]) => {
                     $scope.canEditActiveRepo = $scope.canWriteActiveRepo();
-                    $scope.searchQueries = searchQueries ? searchQueries.data : [];
-                    $scope.allSamples = samples ? samples.data : [];
-                    $scope.usedPrefixes = usedPrefixes;
+                    searchQueries = searchQueriesResponses ? searchQueriesResponses.data : [];
+                    allSamples = samples ? samples.data : [];
+                    usedPrefixes = usedPrefixes;
```
Comment (svilenvelikov) [inline src/js/angular/similarity/controllers/create-index.controller.js:843]: Rename one of the variables

File: test-cypress/integration/explore/similarity.spec.js [hunk 2]
```diff
@@ -238,6 +246,177 @@ describe('Similarity screen validation', () => {
         });
     });
 
+    context('Confirmations when try to change repository', () => {
```
Comment (svilenvelikov) [inline test-cypress/integration/explore/similarity.spec.js:249]: Please, split the tests in these contexts in separate specs

Comment (boyan-tonchev) [reply test-cypress/integration/explore/similarity.spec.js]: Done

File: src/js/angular/similarity/controllers/create-index.controller.js (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/similarity/controllers/create-index.controller.js]: Done

**Comment by boyan-tonchev** [comment]
Jenkins retry build

File: test-cypress/integration/explore/similarity.spec.js [hunk 3]
```diff
@@ -464,4 +582,12 @@ describe('Similarity screen validation', () => {
         const query = 'OPTIONAL { ?result <http://dbpedia.org/ontology/birthPlace> ?birthDate .';
         YasqeSteps.verifyQueryContains(query);
     }
+
+    function createVerifyConfirmationDialogOptions() {
```
Comment (svilenvelikov) [inline test-cypress/integration/explore/similarity.spec.js:586]: I have moved all such functions to respective steps in graphs config tests I'm reworking now

**Review by svilenvelikov — APPROVED** [review]

**Review by svilenvelikov — APPROVED** [review]

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Pull request merged**